Docker/validate_sc2.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+
+"""Validate SC2."""
+
+import argparse
+import json
+
+import dendropy
+
+
+def valid_tree_size(tree):
+    """Check size of prediction tree."""
+
+    return len(tree.taxon_namespace) == 1000
+
+
+def valid_leaf_names(tree):
+    """Check that prediction tree uses correct leaf labels."""
+
+    valid_names = ["x " + "%04d" %
+                   i for i in range(1000)]
+    return all([leaf.label in valid_names
+                for leaf in tree.taxon_namespace])
+
+
+def main(submission, entity_type, results):
+    """Validate submission and write results to JSON.
+
+    Args:
+        submission: input file
+        entity: Synapse entity type
+        results: output file
+    """
+
+    invalid_reasons = []
+
+    if submission is None:
+        invalid_reasons = [
+            f"Expected FileEntity type but found {entity_type}"]
+    else:
+        try:
+            pred_tree = dendropy.Tree.get(file=open(submission, 'r'),
+                                          schema="newick",
+                                          tree_offset=0)
+        except Exception:
+            invalid_reasons = [
+                "Prediction tree not a valid Newick tree format"]
+        else:
+            if not valid_tree_size(pred_tree):
+                invalid_reasons.append(
+                    "Prediction tree does not have 1,000 cell lines")
```

Comment:
Is sc3 10000? (ie. x_10000)?  If so, you can add a parameter that takes in this number and just reuse this for sc3.

Reply:
I had that at first but then I realized the cell identifiers are actually samples (e.g. Run1_Neuron_184301)?

Reply:
I have a feeling you may have to read in the goldstandard for sc3.  Is it possible to read in the goldstandard for sc2 and compare the two trees?  This looks good though.  You can write completely separate validation for sc3.

Reply:
Ooo I like that much better.  I will try it out!